### DISCUSSION TITLE
bulk RNA-seq data trimming

### DISCUSSION THREAD

**Discussion by ChoiJi-Hye** [milaboratory/mixcr#1353] [Q&A] [answered]
Hello,

I have a question regarding using Mixcr with **bulk RNA-seq FASTQ** data, specifically about trimming. 
I apologize if there's already an answer to this question in the past.
 I tried to find previous discussions, but the ones I found were quite old, so I'm posting my question here.

When running Mixcr with RNA-seq FASTQ data, **do I need to perform FASTQ trimming or QC** (Quality Control) separately? 
I am aware that QC results are included in the output files. Should I simply exclude data with poor QC results without prior trimming?

I understand that the manual recommends using FastQC to check the quality of sequences beforehand and that there are trimming options within the tool. However, as this is my first time using this tool and performing TCR analysis, and considering that this tool is versatile for various sequencing types, I'm unsure if I should trim bulk RNA-seq FASTQ data separately or if there are specific settings to configure.

I used the '**mixcr analyze rna-seq**' pipeline that you have provided, but I couldn't see a clear trimming step in it.
**Should trimming be recommended before running Mixcr for bulk RNA-seq FASTQ data?**

I apologize if there is already an answer to this question, and I am repeating it.

Thank you!
Ji-Hye Choi

**Comment by mizraelson** [comment, marked as answer]
Hi,
It's always a good practice to run FASTQC before any analysis to gauge the initial quality of your data. Remember, most bioinformatics software operates on the principle of "garbage in, garbage out." 
If your sequences contain Illumina adapter sequences, it's advisable to use Trimmomatic before executing MiXCR. This capability will be integrated into MiXCR in future versions.
While MiXCR does offer sequence trimming features, it's primarily designed for trimming primers/adapters of known length and/or sequence. Illumina adapters, on the other hand, can appear in varying lengths across reads as artifacts. The `rna-seq` preset in MiXCR does not perform any trimming by default.

**Comment by ChoiJi-Hye** [comment]
Thank you for your prompt and helpful response!

I'm really thankful for the well-crafted tools that enable me to perform the analysis I need.
I'll reanalyze the data after trimming. 

Thanks!
Ji-Hye Choi
